src/OrbitGl/ThreadStateBar.cpp

```diff
@@ -206,9 +208,46 @@ std::string ThreadStateBar::GetThreadStateSliceTooltip(PrimitiveAssembler& primi
   uint64_t end_ns = thread_state_slice->end_timestamp_ns();
   tooltip += absl::StrFormat(
       "<br/>"
-      "<b>Time:</b> %s",
+      "<b>Time:</b> %s"
+      "<br/><br/>",
       orbit_display_formats::GetDisplayTime(TicksToDuration(begin_ns, end_ns)));
 
+  if (thread_state_slice->switch_out_or_wakeup_callstack_id().has_value()) {
+    const orbit_client_data::CallstackData& callstack_data = capture_data_->GetCallstackData();
+    const uint64_t callstack_id = thread_state_slice->switch_out_or_wakeup_callstack_id().value();
+    const orbit_client_data::CallstackInfo* callstack = callstack_data.GetCallstack(callstack_id);
+
+    static const std::string unknown_return_text = "Function call information missing";
+
+    if (thread_state_slice->wakeup_reason() != ThreadStateSliceInfo::WakeupReason::kNotApplicable) {
```

Comment:
Consider adding a comment to explain what it means `kNotApplicable` in the context of this if statement